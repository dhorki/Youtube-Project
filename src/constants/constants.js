export const styles = {
  colors: {
    light: {
      appIcon: '#ff0200',
      background: '#F9F9F9',
      barBackground: '#ffffff',
      primary: '#030303',
      secondary: '#606060',
      linkActive: '#ff0200',
      linkHoverBackground: '#ececec',
      inputBackground: '#ffffff',
      inputBorder: '#cccccc',
      searchButtonBackground: '#f8f8f8',
      // google button
      white: '#fff',
      googleBlue: '#4285f4',
      buttonActiveBlue: '#1669f2',
    },
    dark: {
      appIcon: '#ff0200',
      background: '#181818',
      barBackground: '#212121',
      primary: '#ffffff',
      secondary: '#aaaaaa',
      linkActive: '#ff0200',
      linkHoverBackground: '#303030',
      inputBackground: '#121212',
      inputBorder: '#303030',
      searchButtonBackground: '#323232',
      // google button
      white: '#fff',
      googleBlue: '#4285f4',
      buttonActiveBlue: '#1669f2',
    },
  },
  props: {
    transitionTime: '0.4s',
    sidebarWidth: '150px',
  },
};

export const screenSizes = {
  xsmall: '675px',
  small: '830px',
  mid: '1200px',
};

export const mainViewTypes = {
  searchGallery: 'searchGallery',
  favoritesGallery: 'favoritesGallery',
  videoDetails: 'videoDetails',
};

export const environmentActions = {
  setTheme: 'setTheme',
  setSidebarShow: 'setSidebarShow',
  setSearchQuery: 'setSearchQuery',
  setModalShow: 'setModalShow',
  setUser: 'setUser',
};

export const modalTypes = {
  login: 'login',
  register: 'register',
  resetPassword: 'reset-password',
  none: 'none',
};

export const resetSearchQueryAction = {
  type: environmentActions.setSearchQuery,
  payload: {
    q: '',
  },
};

export const hideModalAction = {
  type: environmentActions.setModalShow,
  payload: {
    modalShow: modalTypes.none,
  },
};
