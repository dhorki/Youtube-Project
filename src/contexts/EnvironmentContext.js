import { createContext } from 'react';

export const EnvironmentContext = createContext();
