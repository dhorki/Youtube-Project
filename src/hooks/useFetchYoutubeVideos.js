import PropTypes from 'prop-types';
import { useFetch } from './useFetch';

export const useFetchYoutubeVideos = (key, idList) => {
  const url =
    'https://www.googleapis.com/youtube/v3/videos?' +
    `key=${key}&` +
    'part=id,snippet,statistics,contentDetails&' +
    `id=${idList.join(',')}&` +
    'maxResults=24';

  const { data, loading, error } = useFetch(url);

  return {
    data,
    loading,
    error,
  };
};

useFetchYoutubeVideos.propTypes = {
  key: PropTypes.string.isRequired,
  idList: PropTypes.arrayOf(PropTypes.string).isRequired,
};
