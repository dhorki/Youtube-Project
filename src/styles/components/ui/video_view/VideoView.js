import styled from 'styled-components';
import { screenSizes, styles } from '../../../../constants/constants';

const videoArea = '60vw';
const videoAreaSmall = '100vw';

export const StyledVideoView = styled.div`
  max-width: ${videoArea};
  min-width: 480px;

  @media only screen and (max-width: ${screenSizes.mid}) {
    max-width: ${videoAreaSmall};
  }

  .embedded-video {
    width: 100%;
    min-width: ${screenSizes.xsmall};
    min-height: calc(480px * 9 / 16);
    height: calc(${videoArea} * 9 / 16);

    @media only screen and (max-width: ${screenSizes.mid}) {
      height: calc(${videoAreaSmall} * 9 / 16);
    }

    iframe {
      width: 100%;
      height: 100%;
      border: none;
    }
  }
  > .view-video-title {
    color: ${({ theme }) => theme.primary};
    font-size: 20px;
    font-weight: bolder;
    margin: 15px 0px;
    padding: 0;
    text-align: left;
    width: 100%;
    min-width: 480px;
  }

  > .view-video-stats {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    color: ${({ theme }) => theme.secondary};
    font-size: 13px;
    font-weight: bolder;
    margin: 0px;
    text-align: left;
    width: 100%;
    min-width: 480px;

    > p {
      margin: 0px;
    }
    > .view-video-stats-right {
      font-size: 15px;

      .live-icon {
        color: ${({ theme }) => theme.appIcon};
      }
    }
  }

  > .view-video-description-box {
    display: flex;
    flex-direction: column;

    > .view-video-description {
      color: ${({ theme }) => theme.secondary};
      overflow: hidden;
      min-height: 100px;
      height: 100%;
    }

    > .view-video-description.collapsed {
      height: 100px;
    }

    > .view-video-description-toggle {
      color: ${({ theme }) => theme.secondary};
      font-weight: bolder;
      background-color: transparent;
      border: none;
      justify-content: left;
      text-align: left;
      margin-top: 30px;
      width: 100px;
      cursor: pointer;
    }
  }

  hr {
    width: 100%;
    border: none;
    border-top: 1px solid ${({ theme }) => theme.inputBorder};
    margin: 15px 0px;
  }

  > .view-video-channel-favorite {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    > button {
      width: 180px;
      height: 40px;
      color: ${styles.colors.dark.primary};
      background-color: ${styles.colors.light.appIcon};
      font-weight: bolder;
      border-radius: 5px;
      border: none;
      cursor: pointer;

      &.remove {
        color: ${styles.colors.dark.primary};
        background-color: ${styles.colors.light.googleBlue};
      }
    }
  }
`;
