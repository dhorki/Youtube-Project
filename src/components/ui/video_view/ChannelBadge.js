import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { StyledChannelBadge } from '../../../styles/components/ui/video_view/ChannelBadge';
import { useFetch } from '../../../hooks/useFetch';
import { intToMagnitude } from '../../../helpers/formatters';
import { EnvironmentContext } from '../../../contexts/EnvironmentContext';

export const ChannelBadge = ({ id }) => {
  const { environment } = useContext(EnvironmentContext);
  const { theme } = environment;

  const url =
    'https://www.googleapis.com/youtube/v3/channels?' +
    `key=${process.env.REACT_APP_YOUTUBE_API_KEY}&` +
    'part=snippet,statistics&' +
    `id=${id}&` +
    'maxResults=24';

  const { data, loading, error } = useFetch(url);

  let image = '',
    title = '#?',
    subscriberCount = 0;
  if (!loading && !error) {
    const item = data?.items[0];
    image = item.snippet.thumbnails.default.url;
    title = item.snippet.title;
    subscriberCount = item.statistics.subscriberCount;
  }

  return (
    <StyledChannelBadge theme={theme}>
      {loading ? null : (
        <div className="badge">
          {!error ? <img src={image} alt={title} /> : null}
          <div>
            <p className="title">{title}</p>
            <p className="subscribers">{intToMagnitude(subscriberCount)} subscribers</p>
          </div>
        </div>
      )}
    </StyledChannelBadge>
  );
};

ChannelBadge.propTypes = {
  id: PropTypes.string.isRequired,
};
