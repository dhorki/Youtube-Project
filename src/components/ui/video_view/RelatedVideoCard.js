import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { timeSince } from '../../../helpers/formatters';
import { decodeHtml } from '../../../helpers/decodeHtml';
import { StyledRelatedVideoCard } from '../../../styles/components/ui/video_view/RelatedVideoCard';
import { EnvironmentContext } from '../../../contexts/EnvironmentContext';

export const RelatedVideoCard = ({ video }) => {
  const { environment } = useContext(EnvironmentContext);
  const { theme } = environment;

  const { id, snippet } = video;
  const { videoId } = id;
  const { title, channelTitle, thumbnails, publishTime } = snippet;
  const { url } = thumbnails.medium;
  const d = new Date(publishTime);

  const upperCaseCharsCount = title.length - title.replaceAll(/[A-Z]/g, '').length;
  const limitLength = upperCaseCharsCount > 50 ? 50 : upperCaseCharsCount > 30 ? 60 : 70;

  const fixedTitle =
    title.length >= limitLength ? title.slice(0, limitLength) + '...' : title;

  return (
    <StyledRelatedVideoCard
      className="animate__animated animate__fadeIn"
      theme={theme}
      to={`/${videoId}`}
    >
      <img src={url} alt="video" />
      <div>
        <div className="videocard-title">{decodeHtml(fixedTitle)}</div>
        <div className="videocard-channel-title">
          {decodeHtml(channelTitle)}
          <br />
          {timeSince(d)} ago.
        </div>
      </div>
    </StyledRelatedVideoCard>
  );
};

RelatedVideoCard.propTypes = {
  video: PropTypes.object.isRequired,
};
