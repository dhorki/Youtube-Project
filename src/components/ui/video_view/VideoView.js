import React, { useState } from 'react';
import { RiLiveLine } from 'react-icons/ri';
import { FaThumbsUp } from 'react-icons/fa';
import { IoTimerOutline } from 'react-icons/io5';
import {
  formatYoutubeDate,
  intToMagnitude,
  numberWithCommas,
  youtubeDurationToTime,
} from '../../../helpers/formatters';
import PropTypes from 'prop-types';
import { useFetchYoutubeVideos } from '../../../hooks/useFetchYoutubeVideos';
import { useTheme } from '../../../hooks/useTheme';

import { StyledVideoView } from '../../../styles/components/ui/video_view/VideoView';
import { LoadingAnimation } from '../loading/LoadingAnimation';
import { ChannelBadge } from './ChannelBadge';
import { ErrorText } from '../error/ErrorText';

export const VideoView = ({ id }) => {
  const { getTheme } = useTheme();
  const theme = getTheme();

  const { data, loading, error } = useFetchYoutubeVideos(
    process.env.REACT_APP_YOUTUBE_API_KEY,
    [id]
  );

  const [descriptionCollapsed, setDescriptionCollapsed] = useState(true);

  const iframeAllow =
    'accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture';

  // const { items } = require('../../../mocks/youtube-video-mock.json');
  let snippet, contentDetails, statistics;
  if (!loading && !error) {
    console.log(data);
    const item = data?.items[0];
    snippet = item.snippet;
    contentDetails = item.contentDetails;
    statistics = item.statistics;
  }

  return (
    <StyledVideoView className="animate__animated animate__fadeIn" theme={theme}>
      {loading ? (
        <LoadingAnimation />
      ) : error ? (
        <ErrorText error={error} />
      ) : (
        <>
          <div className="embedded-video">
            <iframe
              title={id}
              src={`//www.youtube.com/embed/${id}`}
              frameBorder="0"
              allow={iframeAllow}
              allowFullScreen="allowfullscreen"
            />
          </div>
          <p className="view-video-title">{snippet.title}</p>
          <div className="view-video-stats">
            <p className="view-video-stats-left">
              <span>{numberWithCommas(statistics.viewCount)}</span> views -{' '}
              <span>{formatYoutubeDate(snippet.publishedAt)}</span>
            </p>
            <p className="view-video-stats-right">
              <span>
                <FaThumbsUp /> {intToMagnitude(statistics.likeCount)}
              </span>{' '}
              <span>
                <IoTimerOutline />{' '}
                {snippet.liveBroadcastContent === 'live' ? (
                  <RiLiveLine className="live-icon" />
                ) : (
                  youtubeDurationToTime(contentDetails.duration)
                )}
              </span>
            </p>
          </div>
          <hr />
          <div className="view-video-channel-favorite">
            <ChannelBadge id={snippet.channelId} />
            <button>ADD TO FAVORITES</button>
          </div>
          <div className="view-video-description-box">
            <p
              className={`view-video-description ${
                descriptionCollapsed ? 'collapsed' : ''
              }`}
              dangerouslySetInnerHTML={{
                __html: snippet.description.replaceAll('\n', '<br />'),
              }}
            />
            <button
              className="view-video-description-toggle"
              onClick={() => setDescriptionCollapsed(!descriptionCollapsed)}
            >
              SHOW {descriptionCollapsed ? 'MORE' : 'LESS'}
            </button>
          </div>
        </>
      )}
    </StyledVideoView>
  );
};

VideoView.propTypes = {
  id: PropTypes.string.isRequired,
};
