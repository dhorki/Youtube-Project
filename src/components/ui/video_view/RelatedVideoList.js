import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { StyledRelatedVideoList } from '../../../styles/components/ui/video_view/RelatedVideoList';
import { useFetch } from '../../../hooks/useFetch';
import { RelatedVideoCard } from './RelatedVideoCard';
import { LoadingAnimation } from '../loading/LoadingAnimation';
import { ErrorText } from '../error/ErrorText';
import { EnvironmentContext } from '../../../contexts/EnvironmentContext';

export const RelatedVideoList = ({ relatedTo }) => {
  const { environment } = useContext(EnvironmentContext);
  const { theme } = environment;

  const url =
    'https://www.googleapis.com/youtube/v3/search?' +
    `key=${process.env.REACT_APP_YOUTUBE_API_KEY}&` +
    'part=snippet&' +
    'type=video&' +
    'order=relevance&' +
    `relatedToVideoId=${relatedTo}&` +
    'maxResults=24';

  const { data, loading, error } = useFetch(url);
  // const { data, loading, error } = { data: [], loading: true, error: null };

  let filteredItems = [];

  if (!loading && !error) {
    filteredItems = data.items?.filter(
      (item) => item.id.kind.includes('youtube#video') && item.snippet
    );
  }

  return (
    <StyledRelatedVideoList theme={theme}>
      <h3>Related Videos</h3>
      {loading ? (
        <LoadingAnimation />
      ) : error ? (
        <ErrorText error={error} />
      ) : !filteredItems || filteredItems.length === 0 ? (
        'No Related videos.'
      ) : (
        filteredItems.map((video) => (
          <RelatedVideoCard key={video.id.videoId} video={video} />
        ))
      )}
    </StyledRelatedVideoList>
  );
};

RelatedVideoList.propTypes = {
  relatedTo: PropTypes.string.isRequired,
};
