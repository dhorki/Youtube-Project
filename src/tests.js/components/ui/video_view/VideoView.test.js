import React from 'react';
import { Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import { styles } from '../../../../constants/constants';
import { EnvironmentContext } from '../../../../contexts/EnvironmentContext';
import { VideoView } from '../../../../components/ui/video_view/VideoView';
import { useFetchYoutubeVideos } from '../../../../hooks/useFetchYoutubeVideos';
import { act } from 'react-dom/test-utils';
jest.mock('../../../../hooks/useFetchYoutubeVideos');

describe('VideoView', () => {
  const environment = {
    theme: styles.colors.light,
    sidebarShow: false,
    q: '',
  };

  it('should render correctly while loading', () => {
    const videoId = 'videoId';

    useFetchYoutubeVideos.mockReturnValue({
      data: {},
      loading: true,
      error: null,
    });

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <VideoView id={videoId} />
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with success fetch and NOT live broadcast', () => {
    const videoId = 'videoId';

    useFetchYoutubeVideos.mockReturnValue({
      data: {
        items: [
          {
            snippet: {
              title: 'video title',
              channelId: 'channelId',
              description: 'video description',
              publishedAt: '2020-01-01T00:00:00.000Z',
              liveBroadcastContent: 'none',
            },
            contentDetails: {
              duration: 'PT1M30S',
            },
            statistics: {
              viewCount: '100',
              likeCount: '10',
            },
          },
        ],
      },
      loading: false,
      error: null,
    });

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <VideoView id={videoId} />
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with success fetch and live broadcast', () => {
    const videoId = 'videoId';

    useFetchYoutubeVideos.mockReturnValue({
      data: {
        items: [
          {
            snippet: {
              title: 'video title',
              channelId: 'channelId',
              description: 'video description',
              publishedAt: '2020-01-01T00:00:00.000Z',
              liveBroadcastContent: 'live',
            },
            contentDetails: {
              duration: 'PT1M30S',
            },
            statistics: {
              viewCount: '100',
              likeCount: '10',
            },
          },
        ],
      },
      loading: false,
      error: null,
    });

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <VideoView id={videoId} />
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle show more/less on click', () => {
    const videoId = 'videoId';

    useFetchYoutubeVideos.mockReturnValue({
      data: {
        items: [
          {
            snippet: {
              title: 'video title',
              channelId: 'channelId',
              description:
                'Voluptate magna Lorem ullamco ex adipisicing non esse aute elit aliqua reprehenderit eiusmod cillum aute. Ullamco minim do nostrud consequat reprehenderit et exercitation occaecat laboris aute consequat culpa do esse. Irure eu consectetur veniam officia exercitation adipisicing in consequat tempor eu labore fugiat ullamco id. Velit culpa aliquip sint ut occaecat duis.',
              publishedAt: '2020-01-01T00:00:00.000Z',
              liveBroadcastContent: 'live',
            },
            contentDetails: {
              duration: 'PT1M30S',
            },
            statistics: {
              viewCount: '100',
              likeCount: '10',
            },
          },
        ],
      },
      loading: false,
      error: null,
    });

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <VideoView id={videoId} />
      </EnvironmentContext.Provider>
    );

    let descriptionBox = wrapper.find('.view-video-description-box').first();

    act(() => {
      wrapper.find('.view-video-description-toggle').first().simulate('click');
    });
  });

  it('should render correctly with fail fetch', () => {
    const videoId = 'videoId';

    useFetchYoutubeVideos.mockReturnValue({
      data: {},
      loading: false,
      error: { error: 'error' },
    });

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <Router history={createBrowserHistory()}>
          <VideoView id={videoId} />
        </Router>
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
