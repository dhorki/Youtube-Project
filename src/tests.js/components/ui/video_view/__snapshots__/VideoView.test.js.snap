// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoView should render correctly while loading 1`] = `
<div
  className="sc-bczRLJ ieozwB animate__animated animate__fadeIn"
>
  <div
    className="sc-gsnTZi gEmEgX"
  >
    <img
      alt="loading"
      src="https://i.pinimg.com/originals/d4/2b/b2/d42bb2f00e1042c990b25f90ada9b0ab.gif"
    />
  </div>
</div>
`;

exports[`VideoView should render correctly with fail fetch 1`] = `
<div
  className="sc-bczRLJ ieozwB animate__animated animate__fadeIn"
>
  <p
    className="animate__animated animate__fadeIn"
  >
    Error: 
    undefined
  </p>
</div>
`;

exports[`VideoView should render correctly with success fetch and NOT live broadcast 1`] = `
<div
  className="sc-bczRLJ ieozwB animate__animated animate__fadeIn"
>
  <div
    className="embedded-video"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen="allowfullscreen"
      frameBorder="0"
      src="//www.youtube.com/embed/videoId"
      title="videoId"
    />
  </div>
  <p
    className="view-video-title"
  >
    video title
  </p>
  <div
    className="view-video-stats"
  >
    <p
      className="view-video-stats-left"
    >
      <span>
        100
      </span>
       views -
       
      <span>
        Dec 31, 19
      </span>
    </p>
    <p
      className="view-video-stats-right"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
            key="0"
          />
        </svg>
         
        10
      </span>
       
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M112.91 128A191.85 191.85 0 0064 254c-1.18 106.35 85.65 193.8 192 194 106.2.2 192-85.83 192-192 0-104.54-83.55-189.61-187.5-192a4.36 4.36 0 00-4.5 4.37V152"
            fill="none"
            key="0"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="32"
          />
          <path
            d="M233.38 278.63l-79-113a8.13 8.13 0 0111.32-11.32l113 79a32.5 32.5 0 01-37.25 53.26 33.21 33.21 0 01-8.07-7.94z"
            key="1"
          />
        </svg>
         
        1:30
      </span>
    </p>
  </div>
  <hr />
  <div
    className="view-video-channel-favorite"
  >
    <div
      className="sc-dkzDqf iVzNIT"
    />
    <button>
      ADD TO FAVORITES
    </button>
  </div>
  <div
    className="view-video-description-box"
  >
    <p
      className="view-video-description collapsed"
      dangerouslySetInnerHTML={
        Object {
          "__html": "video description",
        }
      }
    />
    <button
      className="view-video-description-toggle"
      onClick={[Function]}
    >
      SHOW 
      MORE
    </button>
  </div>
</div>
`;

exports[`VideoView should render correctly with success fetch and live broadcast 1`] = `
<div
  className="sc-bczRLJ ieozwB animate__animated animate__fadeIn"
>
  <div
    className="embedded-video"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen="allowfullscreen"
      frameBorder="0"
      src="//www.youtube.com/embed/videoId"
      title="videoId"
    />
  </div>
  <p
    className="view-video-title"
  >
    video title
  </p>
  <div
    className="view-video-stats"
  >
    <p
      className="view-video-stats-left"
    >
      <span>
        100
      </span>
       views -
       
      <span>
        Dec 31, 19
      </span>
    </p>
    <p
      className="view-video-stats-right"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
            key="0"
          />
        </svg>
         
        10
      </span>
       
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M112.91 128A191.85 191.85 0 0064 254c-1.18 106.35 85.65 193.8 192 194 106.2.2 192-85.83 192-192 0-104.54-83.55-189.61-187.5-192a4.36 4.36 0 00-4.5 4.37V152"
            fill="none"
            key="0"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="32"
          />
          <path
            d="M233.38 278.63l-79-113a8.13 8.13 0 0111.32-11.32l113 79a32.5 32.5 0 01-37.25 53.26 33.21 33.21 0 01-8.07-7.94z"
            key="1"
          />
        </svg>
         
        <svg
          className="live-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            key="0"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              key="0"
            />
            <path
              d="M16 4a1 1 0 0 1 1 1v4.2l5.213-3.65a.5.5 0 0 1 .787.41v12.08a.5.5 0 0 1-.787.41L17 14.8V19a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1h14zm-1 2H3v12h12V6zM7.4 8.829a.4.4 0 0 1 .215.062l4.355 2.772a.4.4 0 0 1 0 .674L7.615 15.11A.4.4 0 0 1 7 14.77V9.23c0-.221.18-.4.4-.4zM21 8.84l-4 2.8v.718l4 2.8V8.84z"
              fillRule="nonzero"
              key="1"
            />
          </g>
        </svg>
      </span>
    </p>
  </div>
  <hr />
  <div
    className="view-video-channel-favorite"
  >
    <div
      className="sc-dkzDqf iVzNIT"
    />
    <button>
      ADD TO FAVORITES
    </button>
  </div>
  <div
    className="view-video-description-box"
  >
    <p
      className="view-video-description collapsed"
      dangerouslySetInnerHTML={
        Object {
          "__html": "video description",
        }
      }
    />
    <button
      className="view-video-description-toggle"
      onClick={[Function]}
    >
      SHOW 
      MORE
    </button>
  </div>
</div>
`;
